scripts/prepare_sector_network.py

```diff
@@ -3291,6 +3296,201 @@ def set_temporal_aggregation(n, opts, solver_name):
     return n
 
 
+def add_enhanced_geothermal(
+    n,
+    egs_potentials,
+    egs_overlap,
+    costs,
+):
+    """
+    Adds EGS potential to model.
+
+    Built in scripts/build_egs_potentials.py
+    """
+
+    if len(spatial.geothermal_heat.nodes) > 1:
+        logger.warning(
+            "'add_enhanced_geothermal' not implemented for multiple geothermal nodes."
+        )
+
+    config = snakemake.config
+
+    # matrix defining the overlap between gridded geothermal potential estimation, and bus regions
+    overlap = pd.read_csv(egs_overlap, index_col=0)
```

Comment:
What we get from doing it here is that we can use normal `.csv` files as inputs to `prepare_sector_network`, otherwise we would have to resort to something less nice like `.json` or some more obscure encoding within a `.csv`. But your call :)